### PR TITLE
Enhancement/acl restrictions

### PR DESCRIPTION
**Pull request by Yunnigg**

### CODE REVIEW

File: pronto_mia/lib/ui/components/navigation_menu/navigation_menu.dart [hunk 1]
```diff
@@ -49,13 +49,18 @@ class NavigationMenu extends StatelessWidget {
       _buildNavigationCategory(
         'Übersicht',
         [
-          ListTile(
-            leading: const Icon(Icons.today),
-            title: const Text('Einsatzpläne'),
-            onTap: () => model.navigateTo(
-              const DeploymentPlanOverviewView(),
+          if (model.data == null ||
```
Comment (Vec7or) [inline pronto_mia/lib/ui/components/navigation_menu/navigation_menu.dart:52]: Why show the menu if `data==null`?

File: pronto_mia/lib/ui/views/department/overview/department_overview_view.dart [hunk 2]
```diff
@@ -14,26 +14,31 @@ class DepartmentOverviewView extends StatelessWidget {
...
           actions: [
-            ActionSpecification(
-              tooltip: 'Abteilung erstellen',
-              icon: const Icon(Icons.post_add),
-              onPressed: () => model.editDepartment(
-                asDialog: getValueForScreenType<bool>(
-                  context: context,
-                  mobile: false,
-                  desktop: true,
+            if (model.currentUser == null ||
```
Comment (Vec7or) [inline pronto_mia/lib/ui/views/department/overview/department_overview_view.dart:22]: Why show the menu if `currentUser == null`?

File: pronto_mia/lib/ui/views/user/edit/user_edit_viewmodel.dart [hunk 3]
```diff
@@ -125,13 +110,43 @@ class UserEditViewModel extends FormViewModel {
       case 'canEditUsers':
         accessControlList.canEditUsers = value;
         if (value) {
+          accessControlList.canViewUsers = true;
+          accessControlList.canViewDepartments = true;
+          accessControlList.canEditDepartments = true;
```
Comment (Vec7or) [inline pronto_mia/lib/ui/views/user/edit/user_edit_viewmodel.dart:115]: Why is this linked to `CanEditUsers`? It is not necessary in order to change a users department.

File: pronto_mia/lib/ui/views/user/edit/user_edit_viewmodel.dart [hunk 4]
```diff
@@ -125,13 +110,43 @@ class UserEditViewModel extends FormViewModel {
...
         break;
       case 'canEditDepartmentUsers':
         accessControlList.canEditDepartmentUsers = value;
         if (value) {
+          if (!accessControlList.canViewUsers) {
+            accessControlList.canViewDepartmentUsers = true;
+          }
+          accessControlList.canViewDepartments = true;
+          accessControlList.canViewOwnDepartment = false;
+          accessControlList.canEditDepartments = true;
```
Comment (Vec7or) [inline pronto_mia/lib/ui/views/user/edit/user_edit_viewmodel.dart:128]: Not necessary

File: pronto_mia/lib/ui/views/user/overview/user_overview_view.dart [hunk 5]
```diff
@@ -16,26 +16,33 @@ class UserOverviewView extends StatelessWidget {
...
           actions: [
-            ActionSpecification(
-              tooltip: 'Benutzer erstellen',
-              icon: const Icon(Icons.person_add),
-              onPressed: () => model.editUser(
-                asDialog: getValueForScreenType<bool>(
-                  context: context,
-                  mobile: false,
-                  desktop: true,
+            if (model.currentUser == null ||
```
Comment (Vec7or) [inline pronto_mia/lib/ui/views/user/overview/user_overview_view.dart:24]: Why show when `null`?

**Review by Vec7or — CHANGES_REQUESTED** [review]

File: pronto_mia/lib/ui/components/navigation_menu/navigation_menu.dart (hunk 1 above)
Comment (Yunnigg) [reply pronto_mia/lib/ui/components/navigation_menu/navigation_menu.dart]: I reversed the condition. If data == null the widget is not shown.

**Review by Vec7or — APPROVED** [review]

**Pull request merged**